testassets/InteropTestsWebsite/TestServiceImpl.cs

```diff
@@ -0,0 +1,122 @@
+using System.Threading.Tasks;
+using Google.Protobuf;
+using Grpc.Core;
+using Grpc.Core.Utils;
+
+namespace Grpc.Testing
+{
+    // Implementation copied from https://github.com/grpc/grpc/blob/master/src/csharp/Grpc.IntegrationTesting/TestServiceImpl.cs
+    public class TestServiceImpl : TestService.TestServiceBase
+    {
+        public override Task<Empty> EmptyCall(Empty request, ServerCallContext context)
+        {
+            return Task.FromResult(new Empty());
+        }
+
+        public override async Task<SimpleResponse> UnaryCall(SimpleRequest request, ServerCallContext context)
+        {
+            await EnsureEchoMetadataAsync(context);
+            EnsureEchoStatus(request.ResponseStatus, context);
+
+            var response = new SimpleResponse { Payload = CreateZerosPayload(request.ResponseSize) };
+            return response;
+        }
+
+        public override async Task StreamingOutputCall(StreamingOutputCallRequest request, IServerStreamWriter<StreamingOutputCallResponse> responseStream, ServerCallContext context)
+        {
+            await EnsureEchoMetadataAsync(context);
+            EnsureEchoStatus(request.ResponseStatus, context);
+
+            foreach (var responseParam in request.ResponseParameters)
+            {
+                var response = new StreamingOutputCallResponse { Payload = CreateZerosPayload(responseParam.Size) };
+                await responseStream.WriteAsync(response);
+            }
+        }
+
+        public override async Task<StreamingInputCallResponse> StreamingInputCall(IAsyncStreamReader<StreamingInputCallRequest> requestStream, ServerCallContext context)
+        {
+            await EnsureEchoMetadataAsync(context);
+
+            int sum = 0;
+            await requestStream.ForEachAsync(request =>
+            {
+                sum += request.Payload.Body.Length;
+                return TaskUtils.CompletedTask;
+            });
+            return new StreamingInputCallResponse { AggregatedPayloadSize = sum };
+        }
+
+        public override async Task FullDuplexCall(IAsyncStreamReader<StreamingOutputCallRequest> requestStream, IServerStreamWriter<StreamingOutputCallResponse> responseStream, ServerCallContext context)
+        {
+            await EnsureEchoMetadataAsync(context);
+
+            await requestStream.ForEachAsync(async request =>
+            {
+                EnsureEchoStatus(request.ResponseStatus, context);
+                foreach (var responseParam in request.ResponseParameters)
+                {
+                    var response = new StreamingOutputCallResponse { Payload = CreateZerosPayload(responseParam.Size) };
+                    await responseStream.WriteAsync(response);
+                }
+            });
+        }
+
+        public override Task HalfDuplexCall(IAsyncStreamReader<StreamingOutputCallRequest> requestStream, IServerStreamWriter<StreamingOutputCallResponse> responseStream, ServerCallContext context)
```

Comment:
Curious what this scenario is for?

Reply:
https://github.com/grpc/grpc/blob/master/doc/interop-test-descriptions.md#server has a good description of server capabilities, but I think this one is actually a legacy implementation that's not used anymore.